tests/app/views/contexts/summary/test_group.py

```diff
@@ -0,0 +1,111 @@
+def fixture_schema():
+    return {
+        'blocks': [
+            {
+                'type': 'Question',
+                'id': 'number-question-one',
+                'question': {
+                    'answers': [
+                        {
+                            'id': 'answer-one',
+                            'mandatory': False,
+                            'type': 'Number',
+                            'label': 'Leave blank',
+                            'default': 0,
+                        }
+                    ],
+                    'id': 'question',
+                    'title': "Don't enter an answer. A default value will be used",
+                    'type': 'General',
+                },
+            },
+            {
+                'type': 'Question',
+                'id': 'number-question-two',
+                'hide_on_section_summary': True,
+                'question': {
+                    'answers': [
+                        {
+                            'id': 'answer-two',
+                            'mandatory': False,
+                            'type': 'Number',
+                            'label': 'Enter a Value',
+                        }
+                    ],
+                    'id': 'question2',
+                    'title': 'Enter an answer. It will be shown on the summary page',
+                    'type': 'General',
+                },
+            },
+            {'type': 'Summary', 'id': 'summary'},
+        ],
+        'id': 'group',
+        'title': 'group-test',
+    }
+
+
+@pytest.fixture(name='path')
+def fixture_path():
+    return [
+        Location('section', 'number-question-one'),
+        Location('section', 'number-question-two'),
+    ]
+
+
+def test_group_serialize_hide_on_summary_true(
+    app, mocker, schema, path
+):  # pylint: disable=unused-argument
+
+    answer_store = mocker.MagicMock()
+    list_store = mocker.MagicMock()
+    metadata = mocker.MagicMock()
+
+    group = Group(
+        schema, path, answer_store, list_store, metadata, None, Location('test', 'test')
+    )
+
+    assert len(group.blocks) == 1
+    assert group.blocks[0]['id'] == 'number-question-one'
+
+
+def test_group_serialize_hide_on_summary_not_set(
+    app, mocker, schema, path
+):  # pylint: disable=unused-argument
+
+    answer_store = mocker.MagicMock()
+    list_store = mocker.MagicMock()
+    metadata = mocker.MagicMock()
+
+    del schema['blocks'][1]['hide_on_section_summary']
+
+    group = Group(
+        schema, path, answer_store, list_store, metadata, None, Location('test', 'test')
+    )
+
+    assert len(group.blocks) == 2
+    assert group.blocks[1]['id'] == 'number-question-two'
+
+
+def test_group_serialize_hide_on_summary_false(
+    app, mocker, schema, path
+):  # pylint: disable=unused-argument
+    answer_store = mocker.MagicMock()
+    list_store = mocker.MagicMock()
+    metadata = mocker.MagicMock()
+
+    schema['blocks'][0]['hide_on_section_summary'] = False
```

Comment:
As discussed, as we would never see in a normal schema "hide_on_section_summary": False, not sure we should do this in a test

Reply:
That's not what I thought we agreed.

There are three scenarios possible:
  - "hide_on_section_summary": true
  - "hide_on_section_summary": false
  - Hide on section summary omitted defaulting to false.

I'll add a further test that covers the third scenario in addition to the above.

Reply:
Scratch that. I've already tested all three scenarios... :man_facepalming: